src/App.tsx

```diff
@@ -1,25 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const totalItems = 42;
+
+  const lastVisibleItem = currentPage * itemsPerPage;
+  const firstVisibleItem = lastVisibleItem - itemsPerPage;
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setItemsPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstVisibleItem + 1} - ${lastVisibleItem > totalItems ? totalItems : lastVisibleItem} of ${totalItems})`}
```

Comment:
The calculation of the range of visible items might show incorrect values if the 'lastVisibleItem' exceeds 'totalItems'. Ensure that the range displayed to the user is always accurate, especially for the last page.